generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts

```diff
@@ -266,6 +278,43 @@ export abstract class AbstractCsharpGeneratorContext<
         return undefined;
     }
 
+    public isOptional(typeReference: TypeReference): boolean {
+        switch (typeReference.type) {
+            case "container":
+                return typeReference.container.type === "optional";
+            case "named": {
+                const typeDeclaration = this.getTypeDeclarationOrThrow(typeReference.typeId);
+                if (typeDeclaration.shape.type === "alias") {
+                    return this.isOptional(typeDeclaration.shape.aliasOf);
+                }
+                return false;
+            }
+            case "unknown":
+                return false;
+            case "primitive":
+                return false;
+        }
+    }
+
+    public getDefaultValueForPrimitive({ primitive }: { primitive: PrimitiveType }): csharp.CodeBlock {
+        return PrimitiveTypeV1._visit<csharp.CodeBlock>(primitive.v1, {
+            integer: () => csharp.codeblock("0"),
+            long: () => csharp.codeblock("0L"),
+            uint: () => csharp.codeblock("0U"),
+            uint64: () => csharp.codeblock("0UL"),
+            float: () => csharp.codeblock("0.0f"),
+            double: () => csharp.codeblock("0.0d"),
+            boolean: () => csharp.codeblock("false"),
+            string: () => csharp.codeblock('""'),
+            date: () => csharp.codeblock("DateOnly.MinValue"),
+            dateTime: () => csharp.codeblock("DateTime.MinValue"),
+            uuid: () => csharp.codeblock('""'),
+            base64: () => csharp.codeblock('""'),
+            bigInteger: () => csharp.codeblock('""'),
+            _other: () => csharp.codeblock("null")
+        });
+    }
```

Comment:
is this proto specific?

Reply:
No, this actually represents the default value for all of these types. We just happen to only use it for Protobuf so far. The alternative is to simply use the `default` helper ([link](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/default-values#default-value-expressions)), but I figured this looked a little more hand-written / natural. 

The only exception is `string` - technically the default value is `null`, but we actually need to use `""` for protobuf.